### ISSUE TITLE
$params in sourceRecords($params) is always empty

### ISSUE THREAD

**Issue by nspyke** [silverstripe/silverstripe-reports#99]
I have a parameter fields defined
```
public function parameterFields()
{
    return new FieldList([
        DropdownField::create(
            'SubscriptionPeriodID',
            'Subscription Period',
            SubscriptionPeriod::get(),
        ),
    ]);
}
```

and when selecting it and clicking filter, the value appears in the URL as expected, but it is not coming through in `$params` of the `sourceRecords($params)` method. The `$params` array is always empty.

This was working in 4.0. I'm now on 4.1.

**Comment by maxime-rainville** [comment]
I've confirmed the problem. I think it got introduce by commit 5b335ad511a9cdef77c1ecb568d0cb92ba2dd2db.

Basically, we're trying to get the GET `filter` attribute using `HTTPRequest::param()` which only returns route parameters.

We should use `HTTPRequest::param()` instead.

**Comment by dhensby** [comment]
@maxime-rainville you mis-wrote which method we need to use instead, I think...

**Comment by maxime-rainville** [comment]
@dhensby Good crab just corrected it.

**Comment by robbieaverill** [comment]
This would probably be fixed by #107 as well

**Comment by robbieaverill** [comment]
Fixed with #107
